### PR TITLE
test cleanup

### PR DESCRIPTION
**Pull request by kadukf**

### CODE REVIEW

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs [hunk 1]
```diff
@@ -15,211 +15,202 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.UnitTests.PartitionManag
...
         private const string InitialContinuationToken = "initial token";
         private const string PartitionId = "partitionId";
-        private readonly ILease lease, leaseChild, leaseChild2;
-        private readonly ILeaseManager leaseManager;
-        private readonly IPartitionSupervisorFactory partitionSupervisorFactory;
-        private readonly IPartitionSynchronizer synchronizer;
-        private readonly PartitionController sut;
 
-        public PartitionControllerSplitTests()
+        private ILease MockLease(string partitionId = null)
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: This should be named something like CreateMockLease.

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs [hunk 2]
```diff
@@ -15,211 +15,202 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.UnitTests.PartitionManag
...
 
-            partitionSupervisorFactory = Mock.Of<IPartitionSupervisorFactory>(f => f.Create(lease) == partitionSupervisor);
+            //act
+            await sut.AddOrUpdateLeaseAsync(lease).ConfigureAwait(false);
 
-            leaseManager = Mock.Of<ILeaseManager>();
-            Mock.Get(leaseManager)
-                .Setup(manager => manager.AcquireAsync(lease))
-                .ReturnsAsync(lease);
+            //assert
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: remove empty line

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs [hunk 3]
```diff
@@ -15,211 +15,202 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.UnitTests.PartitionManag
...
-            Mock.Get(leaseManager)
-                .Setup(manager => manager.DeleteAsync(lease))
-                .Returns(Task.FromResult(false));
+            var sut = new PartitionController(leaseManager, partitionSupervisorFactory, synchronizer);
 
-            Mock.Get(leaseManager)
-                .Setup(manager => manager.ReleaseAsync(leaseChild))
-                .Returns(Task.FromResult(false));
+            //act
+            await sut.AddOrUpdateLeaseAsync(lease).ConfigureAwait(false);
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: remove 2 empty lines.

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs [hunk 4]
```diff
@@ -15,211 +15,202 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.UnitTests.PartitionManag
...
-                .Setup(s => s.SplitPartitionAsync(It.Is<ILease>(l => l.PartitionId == PartitionId && l.ContinuationToken == LastContinuationToken)))
-                .ReturnsAsync(new[] { leaseChild, leaseChild2 });
+            await sut.ShutdownAsync().ConfigureAwait(false);
 
-            await sut.AddOrUpdateLeaseAsync(lease).ConfigureAwait(false);
-
-            Mock.Get(leaseChild)
-                .VerifySet(l => l.Properties = customProperties, Times.Once);
+            Mock.Get(leaseChild1)
+                    .VerifySet(l => l.Properties = customProperties, Times.Once);
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: fix indent.

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs [hunk 5]
```diff
@@ -15,211 +15,202 @@ namespace Microsoft.Azure.Documents.ChangeFeedProcessor.UnitTests.PartitionManag
...
+            var synchronizer = Mock.Of<IPartitionSynchronizer>();
+            Mock.Get(synchronizer)
+                .Setup(s => s.SplitPartitionAsync(lease))
+                .ReturnsAsync(new[] { MockLease(), MockLease() });
 
-            leaseChild2 = Mock.Of<ILease>();
-            Mock.Get(leaseChild2)
-                .Setup(l => l.PartitionId)
-                .Returns("childPartitionId2");
+            var partitionSupervisor = Mock.Of<IPartitionSupervisor>(o => o.RunAsync(It.IsAny<CancellationToken>()) == Task.FromException(new PartitionSplitException("message", LastContinuationToken)));
```
Comment (mkolt) [inline src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: Looks like these 3 lines are repeated, with some variation. Consider a helper method.

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs (hunk 1 above)
Comment (kadukf) [reply src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: done

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs (hunk 5 above)
Comment (kadukf) [reply src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: I was trying but then the test will be less readable as you need to jump up/down to the method + I'd like to show the reader of the code that RynAsync is throwing exception .. so keeping it as it is .. we can change it in future .

File: src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs (hunk 2 above)
Comment (kadukf) [reply src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionControllerSplitTests.cs]: done

**Pull request merged**